src/basicEventScheduler.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on June 24, 2020
+
+@author: mandd
+"""
+
+#External Modules---------------------------------------------------------------
+import numpy as np
+import xarray as xr
+import pandas as pd
+#External Modules End-----------------------------------------------------------
+
+#Internal Modules---------------------------------------------------------------
+from PluginBaseClasses.ExternalModelPluginBase import ExternalModelPluginBase
+#Internal Modules End-----------------------------------------------------------
+
+class basicEventScheduler(ExternalModelPluginBase):
+  """
+    This class is designed to create a Maintenance Scheduler model
+  """
+  def __init__(self):
+    """
+      Constructor
+      @ In, None
+      @ Out, None
+    """
+    ExternalModelPluginBase.__init__(self)
+
+  def initialize(self, container, runInfoDict, inputFiles):
+    """
+      Method to initialize the Basic Event Scheduler model
+      @ In, container, object, self-like object where all the variables can be stored
+      @ In, runInfoDict, dict, dictionary containing all the RunInfo parameters (XML node <RunInfo>)
+      @ In, inputFiles, list, list of input files (if any)
+      @ Out, None
+    """
+
+  def _readMoreXML(self, container, xmlNode):
+    """
+      Method to read the portion of the XML that belongs to the Basic Event Scheduler model
+      @ In, container, object, self-like object where all the variables can be stored
+      @ In, xmlNode, xml.etree.ElementTree.Element, XML node that needs to be read
+      @ Out, None
+    """
+    container.basicEvents = {}
+
+    for child in xmlNode:
+      if child.tag == 'BE':
+        container.basicEvents[child.text.strip()] = [child.get('tin'),child.get('tfin')]
+      elif child.tag == 'timeID':
+        container.timeID = child.text.strip()
+      elif child.tag == 'variables':
+        variables = [str(var.strip()) for var in child.text.split(",")]
```

Comment:
The variables is not directly used in this class, you probably can remove these lines.

Reply:
The lines are needed as part of the external model class which require the "variables" node

Reply:
The variables are processed in ExternalModel class, but you are not using it in your class.

Reply:
You are right. We are looping over all nodes, even if they have already processed by other class. I'm ok to keep it here.